src/main/java/com/puppycrawl/tools/checkstyle/filters/JavaPatchFilterElement.java

```diff
@@ -43,12 +45,43 @@
      */
     private static final String MAX = "max";
 
+    /**
+     * Map that mapping between a check and its ancestor token types.
+     */
+    private static final Map<String, List<Integer>> CHECK_TO_ANCESTOR_NODES_MAP =
+            new HashMap<String, List<Integer>>() {
+        {
+            put("ArrayTrailingComma", Arrays.asList(TokenTypes.METHOD_DEF));
```

Comment:
What if array was defined in a field or constructor?
Also, under context strategy this check should detect only changes inside an array definition, other statements in method should not trigger its violations. METHOD_DEF is a big scope for it. Please find the closest node which covers whole array in AST.

Reply:
TokenTypes.VARIABLE_DEF is ok

Reply:
What if array is defined without a variable? For example:
```java
       toStream(new int[]{
         1,
         2,
         3 
       });
```
I think you need **ARRAY_INIT** node. AST example: http://rveach.no-ip.org/checkstyle/checkstyle.php?action=view&config=72fba57f86eaf7df54f418b381c28422&code=5d8cb1b4dd4dc06485c792175f91b92e&checkstyle=checkstyle-8.35-all.jar&printTree=true&printSuppressions=

Reply:
ok, done.